### ISSUE TITLE
Code on the landing page

### ISSUE THREAD

**Issue by MakisH** [precice/precice.github.io#469]
Our landing page currently mostly has text and figures (and a video), but does not show any code. I also think that with a code example, everything becomes clearer.

A nice example of this approach is pyiron: https://pyiron.org/

We could insert some (pseudo)code, demonstrating the main idea and potentially some features (e.g., that the API calls are the same independent of the coupling algorithm). We could even use tabs to showcase different languages (see #211), or even additionally link to an interactive platform such as mybinder or codeocean.

I would like to work on this (but it's low priority).

**Comment by Aayushman-nvm** [comment]
Hi @MakisH is this issue open to contributors?

**Comment by MakisH** [comment]
@Aayushman-nvm suggestions are welcome, but it would need some deeper knowledge of what preCICE does, to highlight the important elements. As such, I don't think it is a good idea for anyone just starting. I would suggest discussing concrete suggestions before implementing anything.

I adjusted the labels to highlight that this is more content-oriented than technicalities-oriented (the technicalities we already know how to do, the if/what is the main question).

**Comment by Aayushman-nvm** [comment]
Yep you’re right @MakisH  It makes sense for someone with deeper preCICE experience to decide which snippets/elements best highlight the core concepts.
Once there’s clarity on the direction, I’d be happy to help with the frontend implementation wherever needed. If anything else comes to mind, I’ll be glad to share it as well.

**Comment by Flamki** [comment]
Hi @MakisH, I reviewed the current landing page section and the related “couple your code” docs, and I’d like to align on content direction before implementation.

Proposed content for the existing “How does it look like?” section:
- Add one short pseudo-code snippet showing the typical participant loop (`initialize -> getMaxTimeStepSize -> readData -> solve -> writeData -> advance -> finalize`).
- Add one tiny XML snippet showing that serial/parallel is selected in configuration, while the solver-side API flow stays the same.
- Keep the current asciinema demo and existing links.

Reason for this choice:
- It directly answers the issue (show code on landing).
- It reflects the structure already used in the current docs, so it stays consistent.
- It keeps the landing page minimal and easy to maintain (no new JS/components).

If this content direction looks right, I can implement it as a small content-only PR touching `content/index.html` only, with a screenshot.

**Comment by alankritdabral** [comment]
https://github.com/user-attachments/assets/b3d4e75f-a948-4d96-944d-65489f36f03c

Hi @MakisH,

I’ve created a quick demo video of a possible landing page concept, based on the discussions in this issue along with some of my own ideas.

I’d really appreciate any feedback or suggestions on how it looks, and whether you think this direction makes sense. I’m especially interested in knowing what could be improved or refined further.

Thanks!

**Comment by MakisH** [comment]
Thank you for the prototype, @alankritdabral. I think that the placement looks interesting (I would generally place it lower down the page, but this might be a valid place as well), the style is rather basic (but it's a starting point), the animation looks cool but does not really help, and the content needs more iterations to be consistent with the rest of the website (at least variable names and comments).

As I wrote above:

> suggestions are welcome, but it would need some deeper knowledge of what preCICE does, to highlight the important elements. As such, I don't think it is a good idea for anyone just starting.

In general, I don't think that any `content`-marked issues are a good fit for anyone that is not already using the software.
